### PR TITLE
DOC: Better doc for `jnp.trace`

### PR DESCRIPTION
**Pull request by rajasekharporeddy**
Part of #21461

### CODE REVIEW

File: jax/_src/numpy/lax_numpy.py [hunk 1]
```diff
@@ -6610,10 +6610,53 @@ def triu(m: ArrayLike, k: int = 0) -> Array:
...
+    axis1: optional, default=0. The first axis along which to take the sum of
+      diagonal. Must be a static integer value.
+    axis2: optional, default=1. The second axis along which to take the sum of
+      diagonal. Must be a static integer value.
+    dtype: optional. The dtype of the output array. Should be provided as static
+      argument in JIT compilation.
+    out: Not used by JAX.
+
+  Returns:
+    A 0-D array for 2-D input, and in general a N-2 dimensional array for
```
Comment (jakevdp) [inline jax/_src/numpy/lax_numpy.py:6633]: Maybe we could be more explicit and say this is an array of shape `a.shape[:-2]`.

Comment (rajasekharporeddy) [reply jax/_src/numpy/lax_numpy.py:6633]: I think the shape of the output array may not always be `a.shape[:-2]`, particularly when the `axis1` and `axis2` arguments are modified from their default values. For example:

```python
>>> x = jnp.arange(1, 13).reshape(2, 3, 2)
>>> jnp.trace(x, axis1=0, axis2=2).shape
(3,)
>>> x.shape[:-2]
(2,)
```

Comment (jakevdp) [reply jax/_src/numpy/lax_numpy.py:6633]: OK, in that case, we can say the shape is `x.shape` with the dimensions at `axis1` and `axis` removed.

Comment (rajasekharporeddy) [reply jax/_src/numpy/lax_numpy.py:6633]: Can we write it as "An array of dimension `x.ndim-2` containing the sum of the diagonal elements along axes `(axis1, axis2)`"

Comment (jakevdp) [reply jax/_src/numpy/lax_numpy.py:6633]: Sure, that sounds good.

Comment (jakevdp) [reply jax/_src/numpy/lax_numpy.py:6633]: Side-note: I just noticed the function does not error if `axis1 == axis2`, and it probably should.

Comment (rajasekharporeddy) [reply jax/_src/numpy/lax_numpy.py:6633]: Thanks! I will open another PR to raise an error when `axis1 ==  axis2`.

**Review by jakevdp — APPROVED** [review]
Thanks!

**Comment by jakevdp** [comment]
Merge was prevented by a conflict. Can you rebase against the updated main branch, and we'll try again? Thanks!

**Comment by rajasekharporeddy** [comment]
Thanks! Rebased onto latest upstream main.
